pkg/network/pod/annotations/generator_test.go

```diff
@@ -723,6 +723,116 @@ var _ = Describe("Annotations Generator", func() {
 			Expect(annotations).To(HaveKeyWithValue(networkv1.NetworkAttachmentAnnot, ""))
 		})
 	})
+
+	Context("Unknown fields added by external entities in multus annotation", func() {
+		const (
+			network1Name                     = "red"
+			networkAttachmentDefinitionName1 = "some-net"
+			network2Name                     = "blue"
+			networkAttachmentDefinitionName2 = "other-net"
+
+			multusNetworksAnnotationWithOneSecondaryNetAndUnknownFields = `[` +
+				`{"name":"some-net","namespace":"default","interface":"podb1f51a511f1","mac":"00:1A:2B:3C:4D:5E","foo":"bar","foo1":"bar1"}]`
+
+			multusNetworksAnnotationWithTwoSecondaryNetAndUnknownFields = `[` +
+				`{"name":"some-net","namespace":"default","interface":"podb1f51a511f1","mac":"00:1A:2B:3C:4D:5E","foo":"bar","foo1":"bar1"},` +
+				`{"name":"other-net","namespace":"default","interface":"pod16477688c0e"}]`
+		)
+
+		It("Should preserve unknown fields for for VM with one secondary interface", func() {
+			vmi := libvmi.New(
+				libvmi.WithNamespace(testNamespace),
+				libvmi.WithInterface(*v1.DefaultBridgeNetworkInterface()),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(network1Name)),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithNetwork(libvmi.MultusNetwork(network1Name, networkAttachmentDefinitionName1)),
+				libvmistatus.WithStatus(
+					libvmistatus.New(
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{Name: "default", PodInterfaceName: "eth0"}),
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       network1Name,
+							InfoSource: vmispec.InfoSourceMultusStatus,
```

Comment:
Please also add the domain info source so it will be clearer that this is a running VM.